src/plugins/intel_gna/backend/make_pwl.cpp

```diff
@@ -266,20 +266,35 @@ void make_gna_pwl(const DnnActivation&  fun,
             gna_pwl[0].slope = 0;
             print_segment(gna_pwl[0].xBase / in_scale, gna_pwl[0].yBase / out_scale, 0.0);
 
-            gna_pwl[1].xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
-            gna_pwl[1].yBase = y_lower;
+            int32_t xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
+            int16_t yBase = y_lower;
             s = gna_slope(1.0, in_scale, out_scale);
-            gna_pwl[1].slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
-            gna_pwl[1].xBase = gna_pwl[1].xBase | s.slope_scale_index;
-            print_segment((int32_t)(gna_pwl[1].xBase & XBASEMASK) / in_scale, gna_pwl[1].yBase / out_scale, 1.0);
+            int16_t slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
+
+            gna_pwl[1].xBase = xBase | s.slope_scale_index;
+            gna_pwl[1].yBase = yBase;
+            gna_pwl[1].slope = slope;
+            print_segment(static_cast<int32_t>(gna_pwl[1].xBase & XBASEMASK) / in_scale,
+                          gna_pwl[1].yBase / out_scale,
+                          1.0);
+
+            // It was decided based on 89164 to introduce extra segment to ensure that PWL will cross (0,0).
+            // count zero of function for middle segment.
+            auto x0 = -yBase / s.slope + xBase;
+            // ?? maybe it should be checked with some epsilon?
+            if (x0 != 0.0) {
```

Comment:
I guess `x0` is integer type, so no need to check against `float` ie `0.0` or use epsilon

Reply:
Unfortunately result is double due the fact . s.slope is double.
If whole expression is converted to integer at the end value it is evaluated to 0(for reproduction model) and no segment will be added.